deprecation.py

```diff
@@ -155,8 +158,14 @@ def deprecated(deprecated_in=None, removed_in=None, current_version=None,
     is_unsupported = False
 
     # StrictVersion won't take a None or a "", so make whatever goes to it
-    # is at least *something*.
-    if current_version:
+    # is at least *something*. Compare versions only if removed_in is not
+    # of type datetime.date
+    if isinstance(removed_in, date):
+        if date.today() >= removed_in:
+            is_unsupported = True
+        else:
+            is_deprecated = True
+    elif current_version and not isinstance(removed_in, date):
```

Comment:
@aakum03 : You don't need the `and not` because if `isinstance(removed_in, date)` is `True`, the first `if` branch would have been taken.

Reply:
Yes, I missed that when I was switch around the branches. I've changed it.